src/org/labkey/filetransfer/FileTransferManager.java

```diff
@@ -54,39 +78,105 @@ public static FileTransferManager get()
+        return webPart.getPropertyMap();
     }
 
-    public PropertyManager.PropertyMap getFileTransferConfig(Container container)
+    public FileTransferProvider getProvider(ViewContext context)
     {
-        return PropertyManager.getProperties(container, FileTransferManager.FILE_TRANSFER_CONFIG_PROPERTIES);
+        return getProvider(getWebPartProperties(context), context);
     }
 
-    public Boolean isMetadataListConfigured(Container container)
+    public FileTransferProvider getProvider(Map<String, String> properties)
     {
-        return !getFileTransferConfig(container).isEmpty();
+        return Registry.get().getProvider(properties.get(FILE_TRANSFER_PROVIDER));
     }
 
-    @Nullable
-    public ListDefinition getMetadataList(Container container)
+    public FileTransferProvider getProvider(Map<String, String> properties, ViewContext context)
     {
-        PropertyManager.PropertyMap map = FileTransferManager.get().getFileTransferConfig(container);
+        return Registry.get().getProvider(context.getContainer(), context.getUser(), properties.get(FILE_TRANSFER_PROVIDER));
+    }
+
+    public TransferEndpoint getSourceEndpoint(Map<String, String> properties)
+    {
+        FileTransferProvider provider = getProvider(properties);
+        if (provider == null)
+            return null;
+
+        FileTransferSettings settings = provider.getSettings();
+        TransferEndpoint endpoint = settings.getEndpoint();
+        endpoint.setPath(properties.get(SOURCE_ENDPOINT_DIRECTORY));
+        return endpoint;
+    }
 
-        if (!map.isEmpty())
+    public TransferEndpoint getSourceEndpoint(ViewContext context)
+    {
+        return getSourceEndpoint(getWebPartProperties(context));
+    }
+
+    public List<String> getFileNames(ViewContext context)
+    {
+        HttpSession session = context.getSession();
+        Container sessionContainer = getContainer(context);
+        if (sessionContainer != null)
+            context.setContainer(getContainer(context));
+        String key = (String) session.getAttribute(DATA_REGION_SELECTION_KEY);
+        Map<String, String> properties = getWebPartProperties(context);
+        ListDefinition listDef = FileTransferManager.get().getMetadataList(properties);
+
+        if (listDef != null)
+        {
+            SimpleFilter filter;
+
+            Set<String> selectedVals = DataRegionSelection.getSelected(context, key, true, false);
+
+            if (listDef.getKeyType() == ListDefinition.KeyType.AutoIncrementInteger || listDef.getKeyType() == ListDefinition.KeyType.Integer)
+            {
+                Set<Integer> selectionIds = new HashSet<>();
+                for (String val : selectedVals)
+                {
+                    selectionIds.add(Integer.parseInt(val));
+                }
+                filter = new SimpleFilter(new SimpleFilter.InClause(FieldKey.fromParts(listDef.getKeyName()), selectionIds));
```

Comment:
Yes, I would think so.  I certainly looked for that when doing this wrong a couple of ways.  I'll check if that's easy to do somewhere and otherwise leave a TODO.